### DISCUSSION TITLE
Controller - Error trying to create reusable component

### DISCUSSION THREAD

**Discussion by gguidotti** [react-hook-form/react-hook-form#7910] [Q&A] [answered]
### Version Number

7.27.1

### Codesandbox/Expo snack

https://codesandbox.io/s/gallant-dawn-jch7r4?file=/src/App.jsx

### Steps to reproduce

1. Open up Codesandbox, it should work fine and you'll see some checkboxes;
2. Comment Controller component (line 92 - 128);
3. Uncomment GroupHookForm component (line 87 - 91);
4. See error: "Cannot read properties of undefined (reading 'array')"

### Expected behaviour

Starting Codesandbox taken from: https://github.com/react-hook-form/react-hook-form/issues/476#issuecomment-976898680

I'm expecting the <GroupHookForm> component to be working just fine. I'm passing down some props to make it reusable (eg. different page with different checkboxes options to be shown), I followed same logic in order to create Select, Search and additional components which are working fine in my project.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
hey @gguidotti this is mostly a usage problem not an issue, i will take a look at it and see what's the incorrect part.

**Reply by bluebill1049** [reply]
hmmm interesting one. looks like it's not strictly related to usage i will re-open the issue and look futher.

**Comment by bluebill1049** [comment]
https://github.com/react-hook-form/react-hook-form/issues/7911

**Comment by bluebill1049** [comment, marked as answer]
oh my god... lol this line 

```diff
- const GroupHookForm = (control, name, options)
+ const GroupHookForm = ({ control, name, options })
```

**Reply by gguidotti** [reply]
Not sure what I'm supposed to say 🙄 thanks for your time, I wonder how I forgot about braces lol

**Reply by bluebill1049** [reply]
all good, i did make such mistake in the pass as well 👍
